src/Approximators.jl

```
@@ -88,6 +109,10 @@ transpose(A::ApproximatorRecipe) = ApproximatorAdjoint(A)
 # Undo the transpose wrapper
 transpose(A::ApproximatorAdjoint{<:ApproximatorRecipe}) = A.parent
 
+Base.size(A::ApproximatorRecipe) = (A.n_rows, A.n_cols)
+Base.size(A::ApproximatorRecipe, i::Int64) = i == 1 ? A.n_rows : A.n_cols
+Base.size(A::ApproximatorAdjoint) = (A.parent.n_cols, A.parent.n_rows)
+Base.size(A::ApproximatorAdjoint, i::Int64) = i == 1 ? A.parent.n_cols : A.parent.n_rows
```

Like `Compressor` this should throw an error if the dimension provided is not `1` or `2`. Maybe match the variable naming used in `Compressors.jl` as well.

---

corrected